runtime/compiler/optimizer/EscapeAnalysis.hpp

```diff
@@ -568,6 +576,8 @@ class TR_EscapeAnalysis : public TR::Optimization
    TR_BitVector              *_notOptimizableLocalStringObjectsValueNumbers;
    TR_BitVector              *_blocksWithFlushOnEntry;
    TR_BitVector              *_visitedNodes;
+   TR_BitVector              *_aliasesOfAllocNode;
+   TR_BitVector              *_aliasesOfAllocNode2;
```

Comment:
This does not seem like a good field name. Could we name it better? Can we add documentation on what this field does exactly? Similarly for others, i.e. `candidate2`

Reply:
@fjeremic Filip, sorry for the long delay in responding.  How about instead of `candidate2` and `_aliasesOfAllocNode2`, we used names like `otherAllocNode` and `_aliasesOfOtherAllocNode`, along with a repetition of Vijay's inline comment describing `_aliasesofAllocNode`?  Something like this:

```
/** 
 * Contains sym refs that are just aliases for a fresh allocation
 * i.e., it is used to track allocations in cases such as
 * ...
 * a = new A() 
 * ...
 * b = a 
 * ...
 * c = b 
 *    
 * In this case a, b and c will all be considered aliases of an alloc node and
 * so a load of any of those sym refs will be treated akin to how the fresh
 * allocation would be treated
 */ 
   TR_BitVector              *_aliasesOfAllocNode;
/**
 * Contains sym refs that are just aliases for a second fresh allocation
 * that is under consideration, as with @ref _aliasesOfAllocNode
 */
   TR_BitVector              *_aliasesOfOtherAllocNode;
```

I know the name _aliasesOfOtherAllocNode doesn't really convey much more about its purpose than _aliasesOfAllocNode2 did, but they're really two vectors tracking aliases of particular candidate allocations that are under consideration at any one point in time.  Do you have any other suggestions?

Perhaps more importantly, we could add a comment over the loop that iterates over `cursor2` - something like this:

```
      // Loop over definitions for this use and over all the candidate allocations
      // If the definition comes directly from a candidate for stack allocation, it's harmless;
      // if it's copied from a variable that's aliased with a candidate for stack allocation that
      // was allocated in the same block, it's harmless
      for (cursor2.SetToFirstOne(); cursor2.Valid(); cursor2.SetToNextOne())
               {
               int32_t defIndex2 = cursor2;
               if (defIndex2 == 0)
                  {
                  allnewsonrhs = false;
                  break;
                  }

               TR::Node *defNode2 = _useDefInfo->getNode(defIndex2);
               TR::Node *firstChild = defNode2->getFirstChild();
               bool rhsIsHarmless = false;
               for (Candidate *candidate2 = _candidates.getFirst(); candidate2; candidate2 = candidate2->getNext())
                  {
                  if (candidate2->_node == firstChild)
                     {
                     rhsIsHarmless = true;
                     break;
```

Does that help?

@vijaysun-omr Vijay, does that capture things accurately?  Any other thoughts?

Reply:
Via private communication, Vijay agreed with those suggested adjustments to variable names and comments.